File: Development/ALL Android Projects/Android/ReadJson/app/src/main/java/com/example/readjson/MainActivity.java
package com.example.readjson;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class MainActivity extends AppCompatActivity {


    TextView tv1, tv2;
    Button btn1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tv1= this.findViewById(R.id.textView);
        tv2= this.findViewById(R.id.textView2);
        btn1= this.findViewById(R.id.button);


        btn1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String url= "https://api.npoint.io/0c3d66accd0afae31463";
                new MyAsyncTask().execute(url);
            }
        });
    }

    class MyAsyncTask extends AsyncTask<String, String, String> {

        @Override
        protected String doInBackground(String... params) {
            try {
                URL url = new URL(params[0]);
                HttpURLConnection urlConnect = (HttpURLConnection) url.openConnection();
                urlConnect.setConnectTimeout(700);
                String dataJsonAsStr = convertStreamToString(urlConnect.getInputStream());
                publishProgress(dataJsonAsStr);
            } catch (Exception ex) {
                // handle exceptions
            }
            return "";
        }

        @Override
        protected void onPreExecute() {
            super.onPreExecute();
        }

        @Override
        protected void onPostExecute(String result) {
            super.onPostExecute(result);
        }

        @Override
        protected void onProgressUpdate(String... values) {
            super.onProgressUpdate(values);
            try {
                JSONObject json = new JSONObject(values[0]);
                JSONObject link = json.getJSONObject("coord");
                String SpeedWind = link.getString("link");
                Boolean isactive = link.getBoolean("is_active");
                tv1.setText("active is: " + isactive );
                tv2.setText("link is: " + SpeedWind );

                if(isactive==true){
                    Intent intent = new Intent(MainActivity.this, MainActivity2.class);
                    startActivity(intent);

                }
            } catch (Exception ex) {
                // handle exceptions
            }
        }

        // method to convert input stream to string
//        private String convertStreamToString(InputStream inputStream) throws IOException {
//            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
//            StringBuilder stringBuilder = new StringBuilder();
//            String line;
//            while ((line = reader.readLine()) != null) {
//                stringBuilder.append(line);
//            }
//            reader.close();
//            return stringBuilder.toString();
//        }
    }

    public String convertStreamToString(InputStream inputStream) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
        String line;
        String allString = "";
        try {
            do {
                line = bufferedReader.readLine();
                if (line != null) {
                    allString += line;
                }
            } while (line != null);
            bufferedReader.close();
        } catch (Exception ex) {
            // handle exceptions
        }
        return allString;
    }


}

File: Development/ALL Android Projects/Android/CheckSunset/app/src/main/java/com/example/checksunset/MainActivity.kt
package com.example.checksunset

import android.os.AsyncTask
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import org.json.JSONObject
import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL

class MainActivity : AppCompatActivity() {
    var tv1:TextView?=null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var btn1:Button = this.findViewById<Button>(R.id.button)
        tv1= this.findViewById<TextView>(R.id.textView)
        var ed1:EditText = this.findViewById<EditText>(R.id.editTextTextPersonName)


        btn1.setOnClickListener(){
            var city = ed1.text.toString()
            val url = "https://api.openweathermap.org/data/2.5/weather?q=$city&appid=cb96f0d033deb03547e4eff4a7e7d88f"
            MyAsncTask().execute(url)
        }
    }


// class for implement the AsyncTask Class's Methodes
    inner class MyAsncTask:AsyncTask<String,String,String>(){
        override fun doInBackground(vararg params: String?): String {
            try{
                var url =URL(params[0]) //Creates a URL object from the String representation.
                var urlConnect =url.openConnection() as HttpURLConnection
                urlConnect.connectTimeout = 700
                val dataJsonAsStr =ConvertStreamToString(urlConnect.inputStream)
                publishProgress(dataJsonAsStr)
            }catch (ex:Exception){}
            return ""
        }

        override fun onPreExecute() {
            super.onPreExecute()
        }

        override fun onPostExecute(result: String?) {
            super.onPostExecute(result)
        }

        override fun onProgressUpdate(vararg values: String?) {
            super.onProgressUpdate(*values)
            try{
                var json= JSONObject(values[0])
                val query=json.getJSONObject("wind")
                val SpeedWind=query.getString("speed")
                tv1!!.text = "speed wind is: "+ SpeedWind+"m/s"
            }catch (ex:Exception){}
        }
    }



    //Methode to convert InputStream to StringStream
    fun ConvertStreamToString(inputStream: InputStream):String{
        val bufferReader= BufferedReader(InputStreamReader(inputStream))
        var line:String
        var AllString:String=""
        try {
            do{
                line=bufferReader.readLine()
                if(line!=null){
                    AllString+=line
                }
            }while (line!=null)
            bufferReader.close()
        }catch (ex:Exception){}
        return AllString
    }


}